### ISSUE TITLE
Tarballs don't work with local stores

### ISSUE THREAD

**Issue by tilpner** [NixOS/nix#2053]
This issue occured with NIX_PATH pointing directly to a tarball, but also when NIX_PATH pointed to a Nix file that eventually uses builtins.fetchTarball

```sh
~ > NIX_PATH=https://nixos.org/channels/nixos-unstable/nixexprs.tar.xz NIX_REMOTE=local?root=/tmp/foo nix build -f '<nixpkgs>' hello -vvvv
evaluating file '/nix/store/4v2ahqxpbn00d9782c5475ajfsg4vcg4-nix-2.0/share/nix/corepkgs/derivation.nix'
locking path '/home/till/.cache/nix/tarballs/126x48x0qfsrfbyniw4876rlmaql2ljxqyqsd0rlx54sjivr0kr8-file'
lock acquired on '/home/till/.cache/nix/tarballs/126x48x0qfsrfbyniw4876rlmaql2ljxqyqsd0rlx54sjivr0kr8-file.lock'
acquiring global GC lock '/tmp/foo/nix/var/nix/gc.lock'
acquiring read lock on '/tmp/foo/nix/var/nix/temproots/17268'
acquiring write lock on '/tmp/foo/nix/var/nix/temproots/17268'
downgrading to read lock on '/tmp/foo/nix/var/nix/temproots/17268'
locking path '/home/till/.cache/nix/tarballs/qh082wlqgk1jhyxr4j88q1gnxsxvl7fc-nixexprs.tar.xz-unpacked'
lock acquired on '/home/till/.cache/nix/tarballs/qh082wlqgk1jhyxr4j88q1gnxsxvl7fc-nixexprs.tar.xz-unpacked.lock'
unpacking 'https://nixos.org/channels/nixos-unstable/nixexprs.tar.xz'...
tar: /nix/store/qh082wlqgk1jhyxr4j88q1gnxsxvl7fc-nixexprs.tar.xz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
lock released on '/home/till/.cache/nix/tarballs/qh082wlqgk1jhyxr4j88q1gnxsxvl7fc-nixexprs.tar.xz-unpacked.lock'
lock released on '/home/till/.cache/nix/tarballs/126x48x0qfsrfbyniw4876rlmaql2ljxqyqsd0rlx54sjivr0kr8-file.lock'
error: program 'tar' failed with exit code 2
download thread shutting down
```

**Comment by cleverca22** [comment]
https://github.com/NixOS/nix/blob/master/src/libstore/download.cc#L722
i think the issue is in this general area

**Comment by edolstra** [comment]
Fixed in 64e486ab63a87b18922bbdb8d2414e74afabb8db.
